canals/component/component.py

```diff
@@ -262,11 +261,10 @@ def _decorate(self, class_):
         _check_run_signature(class_)
 
         # Automatically registers all the init parameters in an instance attribute called `init_parameters`.
-        # See `save_init_params()`.
-        class_.__init__ = save_init_params(class_.__init__)
+        class_.__init__ = _save_init_params(class_.__init__)
 
         # Makes sure the self.defaults dictionary is always present
-        class_.__init__ = init_defaults(class_.__init__)
+        class_.defaults = {}
```

Comment:
Some components may define `self.defaults` in their init. How are tests not failing? For example, how is the line above compatible with https://github.com/deepset-ai/canals/blob/f48d5d8bb63b753a52b29b396b0089922becf612/test/sample_components/test_add_value.py#L34  ?